torchgen/gen_aoti_c_shim.py

```diff
@@ -329,18 +330,24 @@ def get_backend_index_for_aoti(
         )
     ):
         backend_index = backend_indices[dispatch_key]
-    elif backend_indices[DispatchKey.CompositeExplicitAutograd].has_kernel(func):
-        # We need to create C shim wrappers for CompositeExplicitAutograd kernels
-        backend_index = backend_indices[DispatchKey.CompositeExplicitAutograd]
-    elif backend_indices[DispatchKey.CompositeExplicitAutogradNonFunctional].has_kernel(
-        func
-    ):
-        # We need to create C shim wrappers for CompositeExplicitAutogradNonFunctional kernels
-        backend_index = backend_indices[
+    else:
+        # for the extend out-of-tree kernels, we don't need to
+        # duplicatly create C shim wrappers for other dispatch keys
+        if aoti_extend:
```

Comment:
```suggestion
        if aoti_extend and diapatch_key == "XPU":
```

Reply:
> 
The flag `aoti_extend`, which now is refined as `extend_aoti_c_shim`, is by design used for extending the c shim of an existing in-tree/partially-in-tree backend, eg, XPU, those totally out-of-tree backend can generate their c shim normally without this flag.  So here we don't need to specify `if aoti_extend and diapatch_key == "XPU"`